File: Engine/src/main/java/org/etieskrill/engine/graphics/gl/shader/impl/ParticleShader.java
package org.etieskrill.engine.graphics.gl.shader.impl;

import io.github.etieskrill.injection.extension.shader.reflection.ReflectShader;
import org.etieskrill.engine.graphics.gl.shader.ShaderProgram;

import java.util.List;

@ReflectShader(files = "ParticlePointVertex.glsl")
public class ParticleShader extends ShaderProgram {
    public ParticleShader() {
        super(List.of("ParticlePointVertex.glsl"));
    }
}

File: Engine/src/commonMain/kotlin/org/etieskrill/engine/graphics/shader/impl/ParticleShader.kt
package org.etieskrill.engine.graphics.shader.impl

import io.github.etieskrill.injection.extension.shader.reflection.ReflectShader
import org.etieskrill.engine.graphics.shader.Shader

@ReflectShader(files = ["ParticlePointVertex.glsl"])
class ParticleShader : Shader(listOf("shaders/ParticlePointVertex.glsl"))
